File: files/2026-02-08/CSEC 08022026.R
library(tidyverse)
library(readxl)

path <- "2026-02-08/Challenge 102.xlsx"
input <- read_excel(path, range = "B2:C13")
test <- read_excel(path, range = "E3:E3", col_names = FALSE) |> pull()

result <- input %>%
  filter(
    str_detect(
      Status,
      regex("(^|[^a-z])complete(d)?([^a-z]|$)", ignore_case = TRUE)
    ),
    str_detect(Status, regex("(^|[^a-z])ok([^a-z]|$)", ignore_case = TRUE)),
    !str_detect(
      Status,
      regex("not.?ok|risk|hold|incomplete", ignore_case = TRUE)
    )
  ) %>%
  count()

all(result == test)
# [1] TRUE

File: files/2026-02-08/CSEC 08022026.py
import pandas as pd
import re

path = "2026-02-08/Challenge 102.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=11)
test = pd.read_excel(path, usecols="E", skiprows=2, nrows=1, header=None).iloc[0, 0]

def clean_string(s):
    return re.sub(r'[^a-z]+', ' ', s.lower())

res = input[input.Status.apply(lambda s:
    "complete" in clean_string(s)
    and "ok" in clean_string(s)
    and not re.search(r'not ok|risk|hold|incomplete', clean_string(s))
)]

print(len(res) == test)
# True